app/src/main/java/com/example/myspacexdemoapp/ui/launch/LaunchDetailsViewModel.kt

```diff
@@ -15,6 +16,22 @@ class LaunchDetailsViewModel(private val spaceXApi: com.example.spacexdemoapp.ap
     ViewModel() {
     private val _launchMutableLiveData = MutableLiveData<LaunchDetailsViewState>()
     val launchLiveData: LiveData<LaunchDetailsViewState> = _launchMutableLiveData
+    private var id: String? = null
+
+    fun init(bundle: Bundle?) {
```

Comment:
get id from bundle in fragment, because in your case View Model knows about DetailsFragmentArgs